src/main/java/com/shaft/database/DBActions.java

```diff
@@ -0,0 +1,190 @@
+	 * @param username   database username
+	 * @param password   password of database user
+	 */
+
+	public DBActions(String dbType, String dbServerIP, String dbPort, String dbName, String username, String password) {
+		this.dbType = dbType;
+		this.dbServerIP = dbServerIP;
+		this.dbPort = dbPort;
+		this.dbName = dbName;
+		this.username = username;
+		this.password = password;
+	}
+
+	/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+	//////////////////////////////////// [private] Reporting Actions
+	/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	private void passAction(String actionName, String testData, String log) {
+		String message = "Successfully performed action [" + actionName + "].";
+		if (testData != null) {
+			message = message + " With the following test data [" + testData + "].";
+		}
+		ReportManager.log(message);
+		if (log != null) {
+			ReportManager.attachAsStep("Passed", "DB Connection Log", log);
+		}
+	}
+
+	private void passAction(String actionName, String testData) {
+		passAction(actionName, testData, null);
+	}
+
+	private void failAction(String actionName, String testData, String log) {
+		String message = "Failed to perform action [" + actionName + "].";
+		if (testData != null) {
+			message = message + " With the following test data [" + testData + "].";
+		}
+		ReportManager.log(message);
+		if (log != null) {
+			ReportManager.attachAsStep("Failed", "DB Connection Log", log);
+		}
+		Assert.fail(message);
+	}
+
+	private void failAction(String actionName, String testData) {
+		failAction(actionName, testData, null);
+	}
+
+	/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+	//////////////////////////////////// [Public] Core Database Actions
+	/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	/**
+	 * Setup connection to database
+	 * 
+	 * @throws ClassNotFoundException
+	 * @throws SQLException
+	 * 
+	 * 
+	 */
+	
+	private void createConnection() {
+		String connectionString = "";
+		try {
+			switch (dbType.toLowerCase().trim()) {
+
+			case ("mysql"):
+				connectionString = "jdbc:mysql://" + dbServerIP + ":" + dbPort + "/" + dbName;
+				break;
+
+			case ("sqlserver"):
+				connectionString = "jdbc:sqlserver://" + dbServerIP + ":" + dbPort + ";databaseName=" + dbName;
+				break;
+
+			case ("postgresql"):
+				connectionString = "jdbc:postgresql://" + dbServerIP + ":" + dbPort + "/" + dbName;
+				break;
+
+			default:
+				ReportManager.log("Database not supported");
+				failAction("createConnection", dbType);
+				break;
+			}
+			connection = DriverManager.getConnection(connectionString, username, password);
+			passAction("createConnection", connectionString);
```

Comment:
it is a must to replace this "passAction" with discrete logging as discussed.

Reply:
done